### ISSUE TITLE
Flatten a depth-concatenated layer does not work

### ISSUE THREAD

**Issue by hakanardo** [ARM-software/ComputeLibrary#375]
A NEFlattenLayer directly following a NEDepthConcatenateLayer does not work. Here is code to produce the bug:

```C
            #include "arm_compute/runtime/NEON/NEFunctions.h"
            #include "arm_compute/core/Types.h"
            #include "utils/Utils.h"
            #include "utils/GraphUtils.h"
            #include <arm_compute/runtime/SubTensor.h>
            #include <assert.h>

            using namespace arm_compute;
            using namespace utils;
            using namespace graph_utils;
            using namespace std;
        
int main(int argc, const char **argv) {
    Tensor tensor_2;
    Tensor tensor_1;
    NEDepthConcatenateLayer nedepthconcatenatelayer_1;
    Tensor tensor_4;
    NEFlattenLayer neflattenlayer_1;
    Tensor tensor_5;

    const TensorShape tensor_2_shape((unsigned int) 3, (unsigned int) 3, (unsigned int) 4, (unsigned int) 1);
    tensor_2.allocator()->init(TensorInfo(tensor_2_shape, 1, DataType::F32));
    NumPyBinLoader numpybinloader_2("tmp2.npy");
    const TensorShape tensor_1_shape((unsigned int) 3, (unsigned int) 3, (unsigned int) 4, (unsigned int) 1);
    tensor_1.allocator()->init(TensorInfo(tensor_1_shape, 1, DataType::F32));
    NumPyBinLoader numpybinloader_1("tmp1.npy");
    std::vector<arm_compute::ITensor*> nedepthconcatenatelayer_1_list;
    nedepthconcatenatelayer_1_list.push_back(&tensor_1);
    nedepthconcatenatelayer_1_list.push_back(&tensor_2);
    nedepthconcatenatelayer_1.configure(nedepthconcatenatelayer_1_list, &tensor_4);
    auto neflattenlayer_1_len = tensor_4.info()->tensor_shape()[0];
    neflattenlayer_1_len *= tensor_4.info()->tensor_shape()[1];
    neflattenlayer_1_len *= tensor_4.info()->tensor_shape()[2];
    neflattenlayer_1_len *= tensor_4.info()->tensor_shape()[3];
    const TensorShape tensor_5_shape((unsigned int) neflattenlayer_1_len);
    tensor_5.allocator()->init(TensorInfo(tensor_5_shape, 1, DataType::F32));
    neflattenlayer_1.configure(&tensor_4, &tensor_5);
    tensor_2.allocator()->allocate();
    tensor_1.allocator()->allocate();
    tensor_4.allocator()->allocate();
    tensor_5.allocator()->allocate();
    numpybinloader_2.access_tensor(tensor_2);
    numpybinloader_1.access_tensor(tensor_1);

    nedepthconcatenatelayer_1.run();
    neflattenlayer_1.run();

    float temp = *reinterpret_cast<float *>(tensor_5.buffer() + tensor_5.info()->offset_element_in_bytes(Coordinates(36)));
    std::cout << temp << "==" << "2?" << std::endl;
    save_to_npy(tensor_5, "res0.npy", false);
}
```

With the input data generated by

```python
        tmp = np.ones([1,8,3,3], np.float32)
        tmp[:,4:,:,:] *= 2
        np.save("tmp1.npy", tmp[:,:4,:,:])
        np.save("tmp2.npy", tmp[:,4:,:,:])

```

The result becomes

```
[[1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
  0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]]
```
Instead of the expected

```
[[1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
  1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2.
  2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]]
```

**Comment by GeorgeARM** [comment]
Seems to be similar with #365 .
A fixed will be part of the maintenance release soon.

**Comment by GeorgeARM** [comment]
Closing. Should have been resolved in 18.03. Reopen if needed.
